src/app/main.rs

```diff
@@ -489,6 +494,7 @@ where
                 // disabled on purpose
                 //.layer(add_server_id_on_rsp::layer())
                 //.layer(add_remote_ip_on_rsp::layer())
+                .layer(strip_header::request::layer(super::L5D_REQUIRE_ID))
```

Comment:
Just curious: does it make sense for the `require_identity_on_endpoint` layer to strip this header instead of doing it in a separate layer?  Presumably we would always want to strip the header after using it?

Reply:
It probably could. The counter-argument is that it's better to have each layer do exactly one thing so that it's easier to reason about if, for instance, other layers want to act on or report headers...

Reply:
I leaned towards a `strip_header` layer because it's still being explicit about the possibility of the`l5d-require-id` header being present. I think the argument could be made for other `l5d-*` headers, so if it's something we want to do I'd rather cover all those situations in a separate PR!